src/utils/supergraph_config.rs

```diff
@@ -837,31 +870,8 @@ pub(crate) async fn resolve_supergraph_yaml(
         }));
     }
 
-    let mut resolved_supergraph_config: SupergraphConfig = subgraph_definitions.into();
-
-    let mut fed_two_subgraph_names = Vec::new();
-    for subgraph_definition in resolved_supergraph_config.get_subgraph_definitions()? {
```

Comment:
This call to `get_subgraph_definitions()` returns error if anything in the config lacks a routing URL. I could have just changed this to iterate over the `SubgraphConfigs` but I would have to do an assertion that every schema source was of type Sdl. So I thought it would be simpler to just do the parsing on the SDL before we put it in the SupergraphConfig instead of afterwards. (Technically this does mean that some error cases we do some extra work before returning the error that we didn't do before.)